File: ehe/ehe.Web/Modules/DosyaUrunler/UrunEkleme/UrunEklemeColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace ehe.DosyaUrunler.Columns;

[ColumnsScript("DosyaUrunler.UrunEkleme")]
[BasedOnRow(typeof(UrunEklemeRow), CheckNames = true)]
public class UrunEklemeColumns
{
    [EditLink]
    public string UrunAdi { get; set; }
    [EditLink]
    public string UrunAciklamasi { get; set; }
    [EditLink]
    public int UrunStokAdeti { get; set; }
    [EditLink]
    public int UrunFiyati { get; set; }
    [EditLink]
    public string UrunKodu { get; set; }

    [QuickFilter, EditLink]
    public string UrunCesitleriidAd { get; set; }
}

File: ehe/ehe.Web/Modules/ServerTypes/DosyaUrunler/UrunEklemeColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { UrunEklemeRow } from "./UrunEklemeRow";

export interface UrunEklemeColumns {
    UrunAdi: Column<UrunEklemeRow>;
    UrunAciklamasi: Column<UrunEklemeRow>;
    UrunStokAdeti: Column<UrunEklemeRow>;
    UrunFiyati: Column<UrunEklemeRow>;
    UrunKodu: Column<UrunEklemeRow>;
    UrunCesitleriidAd: Column<UrunEklemeRow>;
}

export class UrunEklemeColumns extends ColumnsBase<UrunEklemeRow> {
    static readonly columnsKey = 'DosyaUrunler.UrunEkleme';
    static readonly Fields = fieldsProxy<UrunEklemeColumns>();
}